File: Swift-C++-Testsuite/test/protocols/Swift/MutatingMethod.swift
@main
struct MyMain {

    static func main() {
        var lightSwitch = OnOffSwitch.off
        lightSwitch.toggle()
    }
}

enum OnOffSwitch: Togglable {
    case off, on
    mutating func toggle() {
        switch self {
        case .off:
            self = .on
        case .on:
            self = .off
        }
    }
}

protocol Togglable {
    mutating func toggle()
}

File: AutoTranspiledTests/Swift-to-Cpp/try-0/MutatingMethod.cpp

enum OnOffSwitch : char {
    off,
    on
};

void toggle(OnOffSwitch& lightSwitch) {
    switch (lightSwitch) {
    case OnOffSwitch::off : 
        lightSwitch = OnOffSwitch::on;
        break;
    case OnOffSwitch::on : 
        lightSwitch = OnOffSwitch::off;
        break;
    default : 
        break; 
    }
}

int main() {
    OnOffSwitch lightSwitch = OnOffSwitch::off;
    toggle(lightSwitch); 
    return 0;
}